### PR TITLE
[AutoGeneration] SecurityInsights module (1.2.0)

### PR DESCRIPTION
**Pull request by dicolanl**
## Description

Per feedback on PR https://github.com/Azure/azure-powershell/pull/17286 moving to a dedicated branch

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [X] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md [hunk 1]
```diff
@@ -1,100 +0,0 @@
----
-external help file:
-Module Name: Az.ConfidentialLedger
-online version: https://docs.microsoft.com/powershell/module/az.ConfidentialLedger/new-AzConfidentialLedgerAADBasedSecurityPrincipalObject
-schema: 2.0.0
----
```
Comment (BethanyZhou) [inline src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md:6]: What's the purpose of removing this file? Is it expected? If yes, could you please put this change on another PR?

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Generally looks good to me except one comment.

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+### Example 1: List all Automation Rules
+```powershell
+PS C:\> Get-AzSentinelAutomationRule -ResourceGroupName "myResourceGroupName" -workspaceName "myWorkspaceName"
```
Comment (BethanyZhou) [inline src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md:3]:
```suggestion
Get-AzSentinelAutomationRule -ResourceGroupName "myResourceGroupName" -workspaceName "myWorkspaceName"
```

Could you please remove PS C:\> from your examples to avoid breaking user copy-paste-use experience?

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+DisplayName                 : VIP automation rule
```
Comment (BethanyZhou) [inline src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md:5]: Plus, please split your example code and output with following pattern
\```powershell
{code}
\```
\```output
{code}
\```

File: src/SecurityInsights/readme.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+> see https://aka.ms/autorest
+
+``` yaml
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+# lock the commit
+branch: 59eb5a7f1d09d0be2b80b8497785ffa2d784b5b6
+
+input-file:
+  - https://github.com/Azure/azure-rest-api-specs/tree/main/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/AlertRules.json
```
Comment (BethanyZhou) [inline src/SecurityInsights/readme.md:56]: please use $(repo)/specification/... to simplify input path
```suggestion
  - $(repo)/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2021-09-01-preview/AlertRules.json
```

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md (hunk 1 above)
Comment (dicolanl) [reply src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md]: I dont why this file got changed.  But i cant seem to remove the change from the PR.  any idea how to fix?

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md (hunk 2 above)
Comment (dicolanl) [reply src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md:3]: Fixed in lastest commit

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md (hunk 3 above)
Comment (dicolanl) [reply src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md:5]: fixed in latest commit

File: src/SecurityInsights/readme.md (hunk 4 above)
Comment (dicolanl) [reply src/SecurityInsights/readme.md:56]: fixed in latest commit

File: src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md (hunk 1 above)
Comment (dicolanl) [reply src/ConfidentialLedger/docs/New-AzConfidentialLedgerAADBasedSecurityPrincipalObject.md]: disregard fix.  it was some corruption.  PR no longer has this change

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md (hunk 2 above)
Comment (BethanyZhou) [reply src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md]: Thank you. To apply these changes in docs/*.md, please run ./build-module.ps1 again and commit your changes again

File: src/SecurityInsights/examples/Get-AzSentinelMetadata.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+### Example 1: {{ Add title here }}
+```powershell
+ {{ Add code here }}
+```
+```output
+{{ Add output here }}
```
Comment (BethanyZhou) [inline src/SecurityInsights/examples/Get-AzSentinelMetadata.md:6]: Could you fill all placeholders in examples?

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Generally looks good to me

File: src/SecurityInsights/examples/Get-AzSentinelMetadata.md (hunk 5 above)
Comment (dicolanl) [reply src/SecurityInsights/examples/Get-AzSentinelMetadata.md]: Fixed

File: src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md (hunk 2 above)
Comment (dicolanl) [reply src/SecurityInsights/examples/Get-AzSentinelAutomationRule.md]: Done

**Review by BethanyZhou — APPROVED** [review]

**Comment by BethanyZhou** [comment]
Hi @dicolanl ,

seems like the syntaxes of cmdlets are different with design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1161, just reviewed current design again.

For `New/Update-AzSentinelDataConnector`,

- Change `Indicators`, `Teams`, `Incidents`, `DiscoveryLogs` to their singular form

For `Update-AzSentinelAlertRule`, 
- parameter set 'UpdateViaIdentityFusionMLTI', 'UpdateViaIdentityMicrosoftSecurityIncidentCreation', 'UpdateViaIdentityNRT', 'UpdateViaIdentityUpdateScheduled' have the same mandatory parameters, and both of them are not default parameter set which may cause confusion. 
- please add a mandatory to differentiate them or combine them as one parameter set

`Update-AzSentinelDataConnector`, `Update-AzSentinelDataConnector`, `Update-AzSentinelSetting`, they have similar issue with `Update-AzSentinelAlertRule`,
- please add a mandatory to differentiate their parameter sets or combine all parameter sets as one parameter set

please raise a PR target to [gen/SecurityInsights](https://github.com/Azure/azure-powershell/tree/gen/SecurityInsights) and modify these parameters.
